circle.yml

```diff
@@ -0,0 +1,106 @@
+version: 2
+
+workflows:
+  version: 2
+  default:
+    jobs:
+      - xcode-10
+      - xcode-9
+
+step-library:
+  - &restore-cache
+      restore_cache:
+        keys:
+          - carthage-cache-mbstatic-v1-{{ .Environment.CIRCLE_JOB }}-{{ checksum "Cartfile.resolved" }}
+          - carthage-cache-mbstatic-v1- # used if checksum fails
+
+  - &save-cache
+      save_cache:
+        key: carthage-cache-mbstatic-v1-{{ .Environment.CIRCLE_JOB }}-{{ checksum "Cartfile.resolved" }}
+        paths:
+          - Carthage
+
+  - &prepare
+      run:
+        name: Prepare
+        command: |
+          git submodule sync
+          if (brew outdated | grep carthage > /dev/null); then brew upgrade carthage; fi
+          echo "foo" > ~/.mapbox
+
+  - &install-dependencies
+      run:
+        name: Install Dependencies
+        command: |
+          carthage bootstrap --platform ios --cache-builds --no-use-binaries
+          carthage bootstrap --platform tvos --cache-builds --no-use-binaries
+          carthage bootstrap --platform macos --cache-builds --no-use-binaries
+          carthage bootstrap --platform watchos --cache-builds --no-use-binaries
+
+  - &build-test-MapboxStatic-ios-11
```

Comment:
Rather than duplicate these steps for different OS versions, I see two options:

1. Define the OS versions as variables in the job, like in 6e76d6e9ac655c2e2e517df5353aa79aabcf1fcb.
2. Use `latest` as the OS version (the default, apparently), which would allow someone to only have to upload the Xcode versions.